### PR TITLE
ONNX export API and script for transformers

### PR DESCRIPTION
**Pull request by bfineran**
accessible via `sparseml.transformers.export_onnx --task <TASK> --model_path <MODEL_PATH>`

loads the model, recipe, config, and tokenizer from model directory. creates a fake batch of size 1 from tokenizer that is passed directly to the model for export

### CODE REVIEW

File: src/sparseml/transformers/utils/export.py [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+        inputs,
+        onnx_file_path,
+        convert_qat=convert_qat,
+    )
+
+    return onnx_file_path
+
+
+def parse_args() -> argparse.ArgumentParser:
+    parser = argparse.ArgumentParser(
```
Comment (rahul-tuli) [inline src/sparseml/transformers/utils/export.py:182]: Add doctstring or make function private `_parse_args`

File: src/sparseml/transformers/utils/export.py [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+        model,
+        inputs,
+        onnx_file_path,
+        convert_qat=convert_qat,
+    )
+
+    return onnx_file_path
+
+
+def parse_args() -> argparse.ArgumentParser:
```
Comment (rahul-tuli) [inline src/sparseml/transformers/utils/export.py:181]: nit: return type is `argparse.Namespace`

File: src/sparseml/transformers/utils/export.py [hunk 3]
```diff
@@ -0,0 +1,252 @@
...
+    parser = argparse.ArgumentParser(
+        description="Export a trained transformers model to an ONNX file"
+    )
+
+    parser.add_argument(
+        "--task",
+        type=str,
+        required=True,
+        help="task to create the model for. i.e. mlm, qa, glue, ner",
```
Comment (rahul-tuli) [inline src/sparseml/transformers/utils/export.py:190]: nit: Use title case for first word, `Task`. Or make every help string start with lowercase

File: src/sparseml/transformers/utils/export.py [hunk 4]
```diff
@@ -0,0 +1,252 @@
...
+        help=(
+            "Set True to convert QAT graph exports to fully quantized. Default is True"
+        ),
+    )
+    parser.add_argument(
+        "--finetuning_task",
+        type=str,
+        default=None,
+        help=(
+            "optional finetuning task for text classification and token "
```
Comment (rahul-tuli) [inline src/sparseml/transformers/utils/export.py:220]: nit: Capitalize `O`

File: src/sparseml/transformers/utils/export.py [hunk 5]
```diff
@@ -0,0 +1,252 @@
...
+            "classification exports"
+        ),
+    )
+    parser.add_argument(
+        "--onnx_file_name",
+        type=str,
+        default="model.onnx",
+        help=(
+            "name for exported ONNX file in the model directory. "
```
Comment (rahul-tuli) [inline src/sparseml/transformers/utils/export.py:229]: nit: Capitalize `N`

**Review by rahul-tuli — CHANGES_REQUESTED** [review]

**Review by rahul-tuli — DISMISSED** [review]

**Review by markurtz — APPROVED** [review]

**Review by rahul-tuli — APPROVED** [review]

**Review by markurtz — APPROVED** [review]

**Pull request merged**